### PR TITLE
factory

### PR DESCRIPTION
**Pull request by mateigraura**
This pr adds the the factory used to launch the events notifier. Related pr: [#3128](https://github.com/ElrondNetwork/elrond-go/pull/3218)

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+	"github.com/ElrondNetwork/notifier-go/config"
+	"github.com/ElrondNetwork/notifier-go/proxy"
+)
+
+var (
+	backgroundContextTimeout = 5 * time.Second
+)
+
 func main() {
+	cfg, err := config.LoadConfig()
```
Comment (bogdan-rosianu) [inline cmd/main.go:19]: don't we want this app to support flags? you might consider the `"github.com/urfave/cli"` lib

File: config/config.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package config
+
+import "github.com/ElrondNetwork/elrond-go/core"
+
+const configPath = "./config/config.toml"
```
Comment (bogdan-rosianu) [inline config/config.go:5]: this could have been a flag with a default value

File: config/config.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	ConnectorApi ConnectorApiConfig
+}
+
+type ConnectorApiConfig struct {
+	Port         string
+	HubType      string
+	DispatchType string
+}
+
+func LoadConfig() (*GeneralConfig, error) {
```
Comment (bogdan-rosianu) [inline config/config.go:17]: add comments to exported stuff

File: config/config.toml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[ConnectorApi]
+    # The port on which the Hub listens for subscriptions
+    Port = "5000"
+
+    # The type of the hub. Options: | common | custom:<your_hub_id> |
+    # Used for custom implemenations for subscriptions/events filtering
```
Comment (bogdan-rosianu) [inline config/config.toml:6]: implementations

File: eventNotifier.go [hunk 5]
```diff
@@ -1,26 +1,84 @@
...
+			events = append(events, data.Event{
+				Address:    string(eventHandler.GetAddress()),
+				Identifier: string(eventHandler.GetIdentifier()),
+				Data:       string(eventHandler.GetData()),
+				Topics:     topics,
+			})
+		}
+	}
+
+	log.Debug("extracted events from block logs", "num events", len(events))
```
Comment (bogdan-rosianu) [inline eventNotifier.go:76]: log also the block hash?

File: proxy/client/httpClient.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ElrondNetwork/elrond-go/marshal"
+)
+
+const (
+	contentTypeKey   = "Content-Type"
+	contentTypeValue = "application/json"
+	httpPost         = "POST"
```
Comment (bogdan-rosianu) [inline proxy/client/httpClient.go:16]: there is the `http.MethodPost` constant

File: proxy/handlers/hub.go [hunk 7]
```diff
@@ -0,0 +1,128 @@
+package handlers
+
+import (
+	"github.com/ElrondNetwork/notifier-go/filters"
```
Comment (bogdan-rosianu) [inline proxy/handlers/hub.go:4]: sort imports

Comment (bogdan-rosianu) [reply proxy/handlers/hub.go:4]: remove all empty lines and then run `goimports`

File: proxy/api.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+type webServer struct {
+	router        *gin.Engine
+	notifierHub   dispatcher.Hub
+	generalConfig *config.GeneralConfig
+}
+
+// NewWebServer creates and configures an instance of webServer
+func NewWebServer(generalConfig *config.GeneralConfig) (*webServer, error) {
+	router := gin.Default()
```
Comment (bogdan-rosianu) [inline proxy/api.go:23]: I think you should also setup cors:
`router.Use(cors.Default())`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

**Review by stefanandy — APPROVED** [review]

File: config/config.go (hunk 3 above)
Comment (mateigraura) [reply config/config.go:18]: done

File: config/config.toml (hunk 4 above)
Comment (mateigraura) [reply config/config.toml]: done

File: eventNotifier.go (hunk 5 above)
Comment (mateigraura) [reply eventNotifier.go:76]: added the hash on the first log.

File: proxy/client/httpClient.go (hunk 6 above)
Comment (mateigraura) [reply proxy/client/httpClient.go]: replaced 👍🏻

File: proxy/handlers/hub.go (hunk 7 above)
Comment (mateigraura) [reply proxy/handlers/hub.go]: done

File: proxy/api.go (hunk 8 above)
Comment (mateigraura) [reply proxy/api.go:24]: done

File: cmd/main.go (hunk 1 above)
Comment (mateigraura) [reply cmd/main.go:19]: these will be added in a separate PR

File: config/config.go (hunk 2 above)
Comment (mateigraura) [reply config/config.go:5]: same, this will be added when integrating cli flags

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**